server/EXCEPTIONS.md

```diff
@@ -0,0 +1,192 @@
+# Exception handling in our backend
+
+## Quickstart
+
+To generate errors in the backend that will be passed to the
+frontend, throw a [Nest-defined HttpException](https://docs.nestjs.com/exception-filters#throwing-standard-exceptions), <u>_generally within a service_</u>:
+
+```js
+  throw new HttpException(responseBody, status)
+```
+
+Where...
+  - `status` is one of `HttpStatus` enum codes. E.g. `HttpStatus.UNAUTHORIZED`
+  - `responseBody` argument is an object with the following schema.
+    ```js
+       const responseBody = {
+        code: 'APPLICANT_PORTAL_ERROR_CODE',
+        message: 'Applicant Portal could not find coffee. CRM: CRM ran out of tea.',
+      }
+    ```
+    - The `code` property is developer-defined, and should be a unique, concise,
+    descriptive identifier for the type of error. Users can provide this code to
+    help developers locate the code that yielded the error.
```

Comment:
I think this description conflicts with itself a bit... I don't think it needs to be unique to the `throw new` instance. that might be an `id` in JSON:API errors.

I think the code should attempt to convey what is reasonably inferable from a given point in the code — either through a try/catch block (promise failure or other error), using the the `catch` block's error param _OR_ can be explicitly inferred through logic outside of try/catch, for example:

`if (!contact) throw new HTTPException({ code: 'CONTACT_NOT_FOUND });`

Reply:
Ah okay, gotcha, so like what you're saying in your comment below
https://github.com/NYCPlanning/labs-applicant-portal/pull/554/files#r463195525